src/userspace/exec_monitor.cpp

```diff
@@ -81,4 +104,45 @@ int ExecMonitor::run()
 
 }
 
-int ExecMonitor::export_data() { return 0;} // TODO
\ No newline at end of file
+void ExecMonitor::export_data(process_info *p)
+{
+	if (export_format == STDOUT)
+	{
+		printf("%d\t%d\t%d\t%s\n", p->ppid, p->pid, p->tgid, p->name);
+		return;
+	}
+
+	file.open(filename, std::ios::out | std::ios::app | std::ios::binary);
+	if (!file)
+	{
+		std::cout << "Could not create file at path: " << filename << "\n";
+		std::cout << "Writing to default location: " << default_filename << "\n";
+		file.open(default_filename, std::ios::out | std::ios::app | std::ios::binary);
+		if (!file)
+		{
+			std::cerr << "Could not open file.\n";
+			return;
+		}
+	}
+
+	switch (export_format) {
+	case ExportFormat::CSV:
+		file << p->ppid << "," << p->pid << "," << p->tgid << "," << p->name << "\n";
+		break;
+
+	case ExportFormat::PROTOBUF:
+		ExecOutput exec_output;
+		ProcessInfo *process = exec_output.add_process_info();
+		process->set_ppid(p->ppid);
+		process->set_pid(p->pid);
+		process->set_tgid(p->tgid);
+		process->set_name(p->name);
+		if (!exec_output.SerializeToOstream(&file)) {
```

Comment:
Does this mean that you append a series of ExecOutput which all contain one ProcessInfo to a file? Is this what you want ? the proto definition makes it sounds like what you really want is once ExecOutput which contains a series of ProcessInfo? What is an ExecOutput really supposed to be anyway? This would have deserved some comments to make it clearer what is the difference between an ExecOutput and a ProcessInfo.

Reply:
About this, I followed this tutorial and couldn't get it to work any other way: https://developers.google.com/protocol-buffers/docs/cpptutorial. So they have a an object which contains a list of other object, similarly to what I used: ExecOutput contains one or more objects of type ProcessInfo.